torch/optim/lr_scheduler.py

```diff
@@ -165,38 +165,55 @@ def get_lr(self):
 
 class CosineAnnealingLR(_LRScheduler):
     r"""Set the learning rate of each parameter group using a cosine annealing
-    schedule, where :math:`\eta_{max}` is set to the initial lr and
+    schedule, where :math:`\eta_{max}` is set to the initial lr,
+    :math:`T_{mult}` is the multiplicative factor of T_max and
     :math:`T_{cur}` is the number of epochs since the last restart in SGDR:
 
     .. math::
 
         \eta_t = \eta_{min} + \frac{1}{2}(\eta_{max} - \eta_{min})(1 +
-        \cos(\frac{T_{cur}}{T_{max}}\pi))
+        \cos(\frac{T_{cur}}{T_{max}\times T_{mult}}\pi))
 
     When last_epoch=-1, sets initial lr as lr.
 
     It has been proposed in
-    `SGDR: Stochastic Gradient Descent with Warm Restarts`_. Note that this only
-    implements the cosine annealing part of SGDR, and not the restarts.
+    `SGDR: Stochastic Gradient Descent with Warm Restarts`_.
 
     Args:
         optimizer (Optimizer): Wrapped optimizer.
         T_max (int): Maximum number of iterations.
         eta_min (float): Minimum learning rate. Default: 0.
+        T_mult (int): Multiplicative factor of T_max. Default: 2.
         last_epoch (int): The index of last epoch. Default: -1.
 
     .. _SGDR\: Stochastic Gradient Descent with Warm Restarts:
         https://arxiv.org/abs/1608.03983
     """
 
-    def __init__(self, optimizer, T_max, eta_min=0, last_epoch=-1):
+    def __init__(self, optimizer, T_max, eta_min=0, T_mult=2, last_epoch=-1):
```

Comment:
You can't insert arguments before the end of argument list without breaking backward compatibility. Please move it to the end